src/poetry/repositories/pypi_repository.py

```diff
@@ -42,6 +43,7 @@ def __init__(
 
         self._base_url = url
         self._fallback = fallback
+        self._package_cache: DictCache[dict[str, Any]] = DictCache()
```

Comment:
Same as above (`self._match_cache`), spliting this out into individual child caches might be undesired.